### PR TITLE
INT-1303 - Only map on _key when making any resource relationships with SQL instances

### PR DESCRIPTION
**Pull request by mknoedel**
It might make sense to have these `_types` be all the same in order to more accurately map to what Google Cloud thinks about these SQL instance entities. However, that requires changing managed question and notifying customers. 

The code to do that is https://github.com/JupiterOne/graph-google-cloud/pull/376

...which is quite expensive and a big hassle as we have already created managed questions for this data model. Because of this it is likely better to just stick with the existing types.

### CODE REVIEW

File: src/steps/cloud-asset/index.ts [hunk 1]
```diff
@@ -678,31 +667,10 @@ export async function createBindingToAnyResourceRelationships(
...
-               * enabled, thus only creating iam_binding relationships to resources that have already
-               * been ingested by other integrations.
-               *
-               * This is a BIG problem because we can no longer tell a customer with 100% confidence
-               * that they do not have any insecure resources if they have yet to have an integration
-               * ingest that resource.
-               */
-              skipTargetCreation: true,
+              targetFilterKeys: [type ? ['_type', '_key'] : ['_key']],
+              skipTargetCreation: false,
```
[the remark this answers is not in the document]
Comment (austinkelleher) [reply src/steps/cloud-asset/index.ts:671]: Does adoption work if only the `_key` is specified or does it only work if both `_type` and `_key` are present?

File: src/steps/cloud-asset/index.ts [hunk 2]
```diff
@@ -721,13 +689,13 @@ function buildBindingEntityResourceKey(bindingEntity: BindingEntity) {
...
 
-function isOrganizationalHierarchyResource(resourceType: string) {
+function isOrganizationalHierarchyResource(resourceType?: string) {
   return [
     ORGANIZATION_ENTITY_TYPE,
     FOLDER_ENTITY_TYPE,
     PROJECT_ENTITY_TYPE,
     API_SERVICE_ENTITY_TYPE,
-  ].includes(resourceType);
+  ].includes(resourceType as any);
```
[the remark this answers is not in the document]
Comment (austinkelleher) [reply src/steps/cloud-asset/index.ts:700]: Seems like we could remove this hack by doing something like:

```typescript
function isOrganizationalHierarchyResource(resourceType?: string) {
  return !!resourceType && [
      ORGANIZATION_ENTITY_TYPE,
      FOLDER_ENTITY_TYPE,
      PROJECT_ENTITY_TYPE,
      API_SERVICE_ENTITY_TYPE,
    ].includes(resourceType);
}
```

**Review by austinkelleher — APPROVED** [review]

File: src/steps/cloud-asset/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (mknoedel) [reply src/steps/cloud-asset/index.ts:671]: After checking, it looks like adoption requires a `_type` to work, so I'll just set skipTargetCreation to true when we only have a `_key`

File: src/steps/cloud-asset/index.ts [hunk 3]
```diff
@@ -772,28 +745,19 @@ export async function createApiServiceToAnyResourceRelationships(
...
-      // WARNING! TODO: We can not make these relationships for sqladmin.googleapis.com/Instances
-      // A Google Resource Type of sqladmin.googleapis.com/Instances could be a
-      // google_sql_mysql_instance, a google_sql_postgres_instance, or a
-      // google_sql_sql_server_instance. There is no way to tell at the moment.
-      if (
-        !serviceEntity ||
-        !resourceType ||
-        resourceType === MULTIPLE_J1_TYPES_FOR_RESOURCE_KIND
-      )
-        return;
```
Comment (mknoedel) [inline src/steps/cloud-asset/index.ts:795]: It is no longer necessary to have a resource type so this check almost goes away.

File: src/steps/cloud-asset/index.ts [hunk 4]
```diff
@@ -807,20 +771,18 @@ export async function createApiServiceToAnyResourceRelationships(
...
               _type: API_SERVICE_HAS_ANY_RESOURCE_TYPE,
-              source: serviceEntity,
-              /**
-               * The mapper does not properly remove mapper-created entities at the moment. These
-               * entities will never be cleaned up which will cause duplicates.
-               *
-               * Until this is fixed, we should not create mapped relationships with target creation
-               * enabled, thus only creating iam_binding relationships to resources that have already
-               * been ingested by other integrations.
-               */
```
Comment (mknoedel) [inline src/steps/cloud-asset/index.ts:818]: adoption was fixed in the mapper.

File: src/utils/iamBindings/typeToKeyGeneratorMap.test.ts [hunk 5]
```diff
@@ -80,21 +75,6 @@ const entitiesTypesToSkip = [
...
- *
- * These tests have value right now though, so adding this temporary hack
- * to skip sqladmin instances.
- */
-const HACK_SKIP_UNTIL_SQL_INSTANCES_ARE_FIXED = [
-  SQL_ADMIN_MYSQL_INSTANCE_ENTITY_TYPE,
-  SQL_ADMIN_POSTGRES_INSTANCE_ENTITY_TYPE,
-  SQL_ADMIN_SQL_SERVER_INSTANCE_ENTITY_TYPE,
-]; // will be fixed with https://jupiterone.atlassian.net/browse/INT-1303
-
```
Comment (mknoedel) [inline src/utils/iamBindings/typeToKeyGeneratorMap.test.ts:97]: We can remove this hack now.

**Review by rwaterman — APPROVED** [review]
Discussed running some J1Q1 during testing to avoid any issues that *may* be relevant from  https://github.com/JupiterOne/graph-google-cloud/pull/322. If any queries you decide to run to ensure relationships look sane in the UI are still working, I'm good with it.

**Pull request merged**